File: tests/java/tests-java-junit5-retrofit2/src/test/java/api/ApiTestBase.java
package api;

import allure.AllureOkHttp;
import annotations.Framework;
import config.ConfigReader;
import config.TestConfig;
import retrofit2.Retrofit;
import retrofit2.converter.jackson.JacksonConverterFactory;
import tests.AllureMeta;
import org.junit.jupiter.api.BeforeAll;

@Framework("retrofit2")
public class ApiTestBase extends AllureMeta {

    protected static final TestConfig config = ConfigReader.testConfig;

    protected static ReferenceApi api;

    @BeforeAll
    static void setupRetrofit() {
        api = new Retrofit.Builder()
                .baseUrl(ConfigReader.resolveApiBaseUrl())
                .client(AllureOkHttp.client(config))
                .addConverterFactory(JacksonConverterFactory.create())
                .build()
                .create(ReferenceApi.class);
    }

    protected static String bearer(String token) {
        return "Bearer " + token;
    }
}

File: tests/kotlin/tests-kotlin-junit5-api_request-playwright/src/test/kotlin/api/ApiTestBase.kt
package api

import annotations.Framework
import config.ConfigReader
import config.TestConfig
import org.junit.jupiter.api.BeforeAll
import tests.AllureMeta

@Framework("playwright")
open class ApiTestBase : AllureMeta() {

    companion object {
        @JvmField
        val config: TestConfig = ConfigReader.testConfig

        @JvmStatic
        @BeforeAll
        fun setupApi() {
            PlaywrightHttp.setup(config)
        }
    }
}
